src/IJ_mv/HYPRE_IJMatrix.c

```diff
@@ -219,7 +218,8 @@ HYPRE_IJMatrixSetPrintLevel( HYPRE_IJMatrix matrix,
       return hypre_error_flag;
    }
 
-   hypre_IJMatrixPrintLevel(ijmatrix) = 1;
+   hypre_IJMatrixPrintLevel(ijmatrix) = !!(print_level);
```

Comment:
This is not intuitive.  Why is it not just '= print_level'?  What warning was being generated by '=1'?

Reply:
We had a unused variable warning before. I believe `!!(print_level)` is more precise because it normalizes any non-negative `print_level` to one and negative values to zero. But yes, just `= print_level` would also work, I can change if you would like

Reply:
Ah, of course, it's an unused variable warning.  It doesn't make sense to pass in the `print_level` parameter and not use it.  Does the IJ code check for `1` specifically?  My opinion is that this should be `= print_level` with maybe a check to make sure it's not negative.  However, if that will break the code, then what you are doing works.  It's just not clear why you do it, so writing this differently or adding a comment would be good.

Reply:
Good! The IJ code is just doing checks like `if (print_level)`. So, effectively, the variable is working like a boolean flag. Your suggestion works fine, I will implement it

Reply:
PS: This is an example where Wextra helps identifying that the function doesn't behave as it should: If the user code has a call to `HYPRE_IJMatrixSetPrintLevel`, but the user would like to turn off the printouts, it wouldn't be possible on master... This isn't a critical bug, but it's better to fix it :)

Reply:
Agreed, thanks!

Reply:
Change implemented in the commit [8d4ca42](https://github.com/hypre-space/hypre/pull/896/commits/8d4ca42dd93f8831df38f0e106beab87fc291056)